data-dashboard/pages/signup.tsx

```diff
@@ -90,18 +96,78 @@ function SignUp() {
   });
 
   const [credentials, setCredentials] = useState({
-    name: '',
     email: '',
     password: '',
     confirmPassword: '',
   });
 
+  const [error, setError] = useState({
+    email: '',
+    password: '',
+    confirmPassword: '',
+  });
+
+  const validatePassword = (password: string) => {
```

Comment:
By the way, it seems like these methods don't belong in this file and component. First of all, the component is loaded with too much unrelated logic, you are redefining this method every time this component is re-rendered, which reduces performance. Second, you are making this component really hard to read as it has too much unrelated logic. Consider moving this method outside the component into `@/lib/utils`.